pkg/validate/parallel.go

```diff
@@ -0,0 +1,82 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+)
+
+type ParallelValidator struct {
+	runInParallel []SpinnakerValidator
+}
+
+func (p *ParallelValidator) Validate(spinSvc v1alpha2.SpinnakerServiceInterface, options Options) ValidationResult {
+	var results []ValidationResult
+	for _, v := range p.runInParallel {
+		options.Log.Info(fmt.Sprintf("Running validator %T", v))
+		r := v.Validate(spinSvc, options)
+		results = append(results, r)
+		if r.Error != nil && r.Fatal {
+			options.Log.Info(fmt.Sprintf("Validator %T detected a fatal error, aborting", v))
+			return p.aggregateResults(results)
+		}
+	}
+	return p.aggregateResults(results)
+}
+
+func (p *ParallelValidator) validateAccountsInParallel(accounts []Account, options Options, f func(Account, Options) ValidationResult) ValidationResult {
+	options.Log.Info(fmt.Sprintf("Running validation of %d accounts in parallel", len(accounts)))
+	if len(accounts) == 0 {
+		return ValidationResult{}
+	}
+	resultsChannel := make(chan ValidationResult, len(accounts))
+	abortSignal := make(chan bool)
```

Comment:
How can this be called? In our case we want to stop the validation if `failFast` is true.

Reply:
When there's a fatal error, channels `resultsChannel` and `abortSignal` are closed (https://github.com/armory/spinnaker-operator/pull/25/files/2b551dd53a23a767a72a62cb88b28fb91a25a5c6#diff-ce77105330da5c799c1121e94a8ded20R49), abort channel is used to simply discard the result of any concurrent validator after it finishes. Exposing this channel to stop validations externally can be implemented in a separate user story